### PR TITLE
Fixed Logger.Write creating new lines in LogFile

### PR DESCRIPTION
**Pull request by gurrenm3**
Also fixed DirectoryNotFound in Setup.cs

### CODE REVIEW

**Comment by Sewer56** [comment]
For fixing the issue of `Write()` creating new lines in log files, you probably want to be editing [LogWriter.cs](https://github.com/Reloaded-Project/Reloaded-II/blob/554b1d814940d3cc623f28a6dc5bf395e7910e31/source/Reloaded.Mod.Loader/Logging/LogWriter.cs) instead; the logger itself here has already been working as intended.

Another thing to note is that a user could call `Write()` with `\n` as part of the string, which this solution wouldn't handle correctly. So do fix it up in LogWriter instead :wink:

File: source/Reloaded.Mod.Loader/Logging/Logger.cs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+using System.Text;
```
Comment (Sewer56) [inline source/Reloaded.Mod.Loader/Logging/Logger.cs:1]: No changes needed here, so make sure to clean up :p

File: source/Reloaded.Mod.Loader/Logging/LogWriter.cs [hunk 2]
```diff
@@ -82,18 +86,33 @@ public void Flush()
...
+    }
+
+    [MethodImpl(MethodImplOptions.Synchronized)]
+    private void OnWriteLine(object sender, (string text, Color color) e)
     {
         if (_logItems.Count == MaxBufferLength)
             Flush();
 
-        _logItems.Add($"[{DateTime.Now:HH:mm:ss}] {message}");
+        if(_writeCache.Length == 0)
```
Comment (Sewer56) [inline source/Reloaded.Mod.Loader/Logging/LogWriter.cs:100]: Just add a space here after `if` to meet style guidelines.
Aside from that, everything is a-ok.

**Review by Sewer56 — APPROVED** [review]

File: source/Reloaded.Mod.Loader/Logging/LogWriter.cs (hunk 2 above)
Comment (gurrenm3) [reply source/Reloaded.Mod.Loader/Logging/LogWriter.cs:100]: Oh my bad, missed that

**Pull request merged**